### PR TITLE
PPT-1087 - Contact details - Amend to single input field

### PR DESCRIPTION
**Pull request by SteveSugden**
### Description of Work carried through

Change ContactDetails "Who is the primary contact?" page to have single input field for name
Depends on 
- BE change - https://github.com/hmrc/plastic-packaging-tax-registration/pull/51
- AT change - https://github.com/hmrc/plastic-packaging-tax-acceptance/pull/52

#### Check list 
 - [x] `./precheck` was executed (Integration/Component/Unit tests)
 - [x] `sbt scalafmt test:scalafmt` was executed
 - [ ] Required Environment Config has been amended/added
 - [x] Links to dependencies have been included (BE/FE work)
 - [x] User Acceptance Tests (UAT) were run locally and have passed.
 - [x] No Accessibility errors/warnings reported by Wave

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtax/registration/forms/FullName.scala [hunk 1]
```diff
@@ -20,30 +20,21 @@ import play.api.data.Forms.text
 import play.api.data.{Form, Forms}
 import play.api.libs.json.{Json, OFormat}
 
-case class FullName(firstName: String, lastName: String)
+case class FullName(value: String)
```
Comment (RyanKellyHMRC) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/forms/FullName.scala:23]: Do we still need this `FullName` class or should we refactor to use a String?

File: tampermonkey/PPT_AutoComplete.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // ==UserScript==
 // @name         PPT Registration AutoComplete
 // @namespace    http://tampermonkey.net/
-// @version      14.1
+// @version      14.2
```
Comment (RyanKellyHMRC) [inline tampermonkey/PPT_AutoComplete.js:4]: 😉

File: test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/ContactDetailsFullNameControllerSpec.scala [hunk 3]
```diff
@@ -107,71 +103,29 @@ class ContactDetailsFullNameControllerSpec extends ControllerSpec {
...
-          }
-
-          "last name" in {
-            authorizedUser()
-            val result = controller.submit()(
-              postRequestEncoded(FullName("FirstName", "averyveryveryveryverylongname"), formAction)
-            )
+        "user enters a long name" in {
+          authorizedUser()
+          val result = controller.submit()(postRequestEncoded(FullName("abced" * 40), formAction))
```
Comment (RyanKellyHMRC) [inline test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/ContactDetailsFullNameControllerSpec.scala:116]: You feel comfortable asking the reader to perform `5 * 40` in their poor heads?

**Review by RyanKellyHMRC — COMMENTED** [review]
Wondering whether we ought to remove the `FullName` class?

File: app/uk/gov/hmrc/plasticpackagingtax/registration/views/full_name_page.scala.html [hunk 4]
```diff
@@ -48,26 +47,16 @@
     ))
 
     @govukInput(Input(
-        id    = firstNameField.name,
-        name  = firstNameField.name,
-        value = firstNameField.value,
+        id    = fullNameField.name,
```
Comment (RyanKellyHMRC) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/views/full_name_page.scala.html:50]: Why don't we set the id to `name` and make the business of identifying the correct field in the acceptance tests that bit cleaner?

File: app/uk/gov/hmrc/plasticpackagingtax/registration/forms/FullName.scala (hunk 1 above)
Comment (SteveSugden) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/forms/FullName.scala:23]: This is the "form" class which does the binding from the view, so yes.  It is now analogous to`case class JobTitle(value: String)` or `case class EmailAddress(value: String)`

File: app/uk/gov/hmrc/plasticpackagingtax/registration/views/full_name_page.scala.html (hunk 4 above)
Comment (SteveSugden) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/views/full_name_page.scala.html:50]: Again, just following existing patterns used for JobTitle and EmailAdress

File: test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/ContactDetailsFullNameControllerSpec.scala (hunk 3 above)
Comment (SteveSugden) [reply test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/ContactDetailsFullNameControllerSpec.scala:116]: Would you prefer I defined `val twoHundreadCharacterName = "abcde" * 40` and used that?

Comment (RyanKellyHMRC) [reply test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/ContactDetailsFullNameControllerSpec.scala:116]: Nah, it was a joke. 😁

**Review by RyanKellyHMRC — APPROVED** [review]

**Pull request merged**